cmd/zrepl/zrepl.c

```diff
@@ -78,14 +95,24 @@ open_zvol(int fd, zvol_info_t **zinfopp)
 		hdr.status = ZVOL_OP_STATUS_FAILED;
 		goto open_reply;
 	}
+	if (zinfo->state != ZVOL_INFO_STATE_ONLINE) {
+		LOG_ERR("zvol %s is not online", open_data.volname);
+		hdr.status = ZVOL_OP_STATUS_FAILED;
+		goto open_reply;
+	}
 	zv = zinfo->zv;
-	ASSERT3P(zv, !=, NULL);
-	if (zv->zv_metavolblocksize != 0 &&
-	    zv->zv_metavolblocksize != open_data.tgt_block_size) {
-		LOG_ERR("Conflicting block size");
+
+	if (zv->zv_metavolblocksize != 0) {
+		LOG_ERR("there might be already a data connection for %s",
+		    open_data.volname);
 		hdr.status = ZVOL_OP_STATUS_FAILED;
 		goto open_reply;
 	}
+
+	ASSERT3P(zv, !=, NULL);
+	ASSERT3P(zv->zv_status, ==, ZVOL_STATUS_DEGRADED);
+	ASSERT3P(zv->rebuild_info.zv_rebuild_status, ==, ZVOL_REBUILDING_INIT);
```

Comment:
do we initialize this `zv_rebuild_status` on zinfo init?

Reply:
done. This and above one are initialized as part of bzero, but, changes are done to set them explicit.